drizzle-orm/src/pg-core/dialect.ts

```diff
@@ -153,8 +155,9 @@ export class PgDialect {
 
 				if (is(field, SQL.Aliased) && field.isSelectionField) {
 					chunk.push(sql.identifier(field.fieldAlias));
-				} else if (is(field, SQL.Aliased) || is(field, SQL)) {
-					const query = is(field, SQL.Aliased) ? field.sql : field;
+				} else if (is(field, SQL.Aliased) || is(field, SQL) || is(field, BuiltInFunction)) {
```

Comment:
If we implement the functions as a simple `sql` operator call, this wouldn't be necessary